File: macos-reverse-engineering-10weeks/code/week10/report_builder.cpp
// 10_report_builder.cpp — Khung báo cáo RE + checklist trước khi kết thúc
// Mục tiêu: tổng hợp phát hiện RE (Mach-O, hook, license) + checklist an toàn.
// An toàn: thuần tính toán, chỉ in văn bản.  g++ -std=c++17 -O2 report_builder.cpp -o demo
#include <iostream>
#include <string>
#include <vector>

int main() {
    std::cout << "BÁO CÁO RE macOS — khung\n" << std::string(56,'=') << "\n\n";
    std::cout << "1. Mach-O: kiến trúc + load commands\n";
    std::cout << "2. Static: class-dump + strings + codesign\n";
    std::cout << "3. Dynamic: LLDB/Frida + unified log\n";
    std::cout << "4. Cơ chế license: luồng kiểm tra + hook\n";
    std::cout << "5. Đề xuất phòng thủ: server-side check, anti-tamper\n";

    std::cout << "\nChecklist an toàn:\n";
    std::vector<std::string> ck = {"VM cô lập", "snapshot", "hash trước/sau", "ẩn danh dữ liệu"};
    for (auto& c : ck) std::cout << "  [ ] " << c << "\n";
    return 0;
}

File: macos-reverse-engineering-10weeks/code/week10/report_builder.py
#!/usr/bin/env python3
# ============================================================
# report_builder.py — Sinh khung báo cáo RE tổng hợp
# Mục tiêu: tạo template báo cáo reverse engineering.
# Đầu vào: không.
# Đầu ra: khung báo cáo RE (in ra).
# An toàn: dữ liệu do bạn điền; không chứa secret/mẫu thật.
# ============================================================

def main():
    print("""# Báo cáo Reverse Engineering — macOS

## 1. Tổng quan
- Mục tiêu: <app/mẫu>
- Phạm vi hợp pháp: <app của mình / được cấp phép / mã nguồn mở>

## 2. Static analysis
- Nhận dạng: <hash, file, otool>
- Strings / symbol: <kết quả>
- Decompile (Ghidra): <hàm chính + luồng>

## 3. Dynamic analysis
- LLDB/Frida: <breakpoint, hook, tham số>
- Hành vi: <file, network, process>

## 4. Runtime / hooking
- Swizzle / dylib / Frida: <kỹ thuật dùng>

## 5. Kết luận
- Cơ chế hoạt động: <tóm tắt>
- Điểm yếu/phòng thủ: <nếu là app của bạn>

## 6. Bằng chứng
- <ảnh, log, output script>
""")


if __name__ == "__main__":
    main()
